packages/adelus/src/Adelus_ConfigDefs.h

```diff
@@ -0,0 +1,93 @@
+/*
+//@HEADER
+// ************************************************************************
+//
+//               Pliris: Parallel Dense Solver Package
+//                 Copyright 2004 Sandia Corporation
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef ADELUS_CONFIGDEFS_H
+#define ADELUS_CONFIGDEFS_H
+
+#ifndef __cplusplus
+#define __cplusplus
+#endif // ifndef __cplusplus
+
+/* this section undefines all the things autotools defines for us that we wish it didn't. */
+
+#ifdef PACKAGE
+#undef PACKAGE
+#endif // ifdef PACKAGE
+
+#ifdef PACKAGE_NAME
+#undef PACKAGE_NAME
+#endif // ifdef PACKAGE_NAME
+
+#ifdef PACKAGE_BUGREPORT
+#undef PACKAGE_BUGREPORT
+#endif // ifdef PACKAGE_BUGREPORT
+
+#ifdef PACKAGE_STRING
+#undef PACKAGE_STRING
+#endif // ifdef PACKAGE_STRING
+
+#ifdef PACKAGE_TARNAME
+#undef PACKAGE_TARNAME
+#endif // ifdef PACKAGE_TARNAME
+
+#ifdef PACKAGE_VERSION
+#undef PACKAGE_VERSION
+#endif // ifdef PACKAGE_VERSION
+
+#ifdef VERSION
+#undef VERSION
+#endif // ifdef VERSION
+
+// end of undoing autoconf's work section
+
+#include "Adelus_config.h"
+
+#ifdef HAVE_STRING
+#include <string>
+#else
+#include <string.h>
+#endif
+
+using std::string;
```

Comment:
Does this need to be defined for everyone using Adelus?  Typically using statements in headers is frowned upon because all downstream libraries inherit them.

Reply:
It is not necessary. Right now, only AdelusVersion uses it, so I move it to AdelusVersion.